lib/optimizely/config_manager/polling_config_manager.rb

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2019, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+require_relative '../error_handler'
+require_relative 'base_config_manager'
+require_relative 'exceptions'
+require_relative 'helpers/constants'
+require_relative 'helpers/validator'
+require_relative 'logger'
+require_relative 'notification_center'
+require_relative 'project_config'
+require_relative 'static_config_manager'
+
+require 'net/http'
+
+module Optimizely
+  class PollingConfigManager < StaticConfigManager
```

Comment:
This should not be derived from the `StaticConfigManager`. 

The `StaticConfigManager` should only be a holder for the proejct config and should implement the `BaseConfigManager`

The `PollingConfigManager` is another implementation of the `BaseConfigManager`.